### PR TITLE
#440 build and publis api docs

### PR DESCRIPTION
**Pull request by nimshi89**
#440

### CODE REVIEW

File: source/conf.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = "csvcubed"
+copyright = "2023, ONS"
+author = "ONS"
+release = "0.3.5"
```
Comment (robons) [inline source/conf.py:12]: This needs to be automatically updated when we do a release (via one of the GitHub workflows)

Comment (nimshi89) [reply source/conf.py:12]: yeah okay will sort it out

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	echo "Found $@"
```
Comment (robons) [inline Makefile:20]: We should get rid of this unnecessary line now.

**Comment by SarahJohnsonONS** [comment]
Nick and I had a look, but we're struggling to understand the step-by-step process for generating the docs (e.g. where to set specific modules to be included in the documentation). May be useful to have a chat with Charles/Jozsef so we can understand the process.

**Review by canwaf — APPROVED** [review]
Good comments and a lot of fixing Sphinx.

**Pull request merged**